static/app/components/route.tsx

```diff
@@ -17,6 +17,7 @@ interface BaseRouteObject {
    * Including an `Outlet` component as a child element.
    */
   deprecatedRouteProps?: never;
+  handle?: Record<string, unknown>;
```

Comment:
Don't want to waste time bikeshedding on naming, but the purpose of `handle` seems quite unclear to me. Is this intended to be public or just a temporary thing? If this is going to stick around for a while, I'd expect this to be something like `routeMetadata` or similar.